engine/main.py

```diff
@@ -640,6 +645,28 @@ def unsubscribe_group(group_name, user):
     logging.debug(res)
     return res
 
+def check_unsubscribe(membergroup, user):
+    """Checks whether or not a user can unsubscribe from a mailing list
+
+    Args:
+        membergroup(MemberGroup): a MemberGroup instance of a user who tries to unsubscribe
+        user(UserProfile):a UserProfile instance of the user trying to unsubscribe from membergroup
+
+    Returns:
+        True: if the user is not the last remaining admin in the membergroup
+        False: if the user is the last remainign admin in the membergroup
+    """
+    can_unsubscribe = True
+    admins = []
+
+    for m in membergroup:
+        if m.admin:
+            admins.append(m.member.email.encode("utf-8"))
+    if len(admins) == 1 and user.email in admins:
+        can_unsubscribe = False
```

Comment:
This for loop can be shortened. 
`membergroup.filter(admin=True).values('member__email', flat=True)`
https://docs.djangoproject.com/en/3.0/ref/models/querysets/#values

Reply:
Thanks! Used `values_list('member__email', flat=True)`